ADMVEDU/ViewControllers/FilterViewController.swift

```diff
@@ -18,51 +18,120 @@ class FilterViewController: UIViewController {
         view.backgroundColor = .white
         configurePeakers()
         configureSwitch()
+        configureSlider()
+        configureNavController()
+        setOptions()
+    }
+    
+    func setOptions() {
+        let country = requestOptions.country
+        let media = requestOptions.media
+        let limit = requestOptions.limit
+        let expl = requestOptions.explisit
+
+        var items = Country.allCases.map { $0.rawValue }
+        for index in 0..<items.count {
+            if items[index] == country {
+                countryPicker.selectRow(index, inComponent: 0, animated: false)
+                countryTextField.text = getCountryName(countryCode: country)
+                break
+            }
+        }
 
+        items = MediaType.allCases.map { $0.rawValue }
+        for index in 0..<items.count {
+            if items[index] == media.rawValue {
+                mediaPicker.selectRow(index, inComponent: 0, animated: false)
+                mediaTextField.text = media.rawValue
+                break
+            }
+        }
+
+        explicitSwitch.isOn = (expl == .yes)
+        
+
+        limitSlider.value = Float(limit)
+        sliderLable.text = String(Int(limitSlider.value))
     }
 
-    func configureSwitch() {
-        let tag = getPickerIdentifier()
-        let fieldWidth = CGFloat(300)
-        let fieldHeith = CGFloat(40)
-        let xPoint = view.center.x - fieldWidth/2
-        let yPoint = 100 + 60 * CGFloat(tag)
 
-        tableView.frame = CGRect(x: xPoint, y: yPoint, width: fieldWidth, height: fieldHeith)
+    func configureNavController() {
+        let doneButton = UIBarButtonItem(barButtonSystemItem: .done,
+                                         target: self, action: #selector(applyOptions))
+        let cancelButton = UIBarButtonItem(barButtonSystemItem: .cancel,
+                                           target: self, action: #selector(cancelOptions))
+        navigationItem.rightBarButtonItem = doneButton
+        navigationItem.leftBarButtonItem = cancelButton
+    }
 
+    @objc
+    func applyOptions() {
+        navigationController?.popViewController(animated: true)
+        delegate?.setOptions(with: requestOptions)
+        dismiss(animated: true, completion: nil)
+    }
+
+    @objc
+    func cancelOptions() {
+        navigationController?.popViewController(animated: true)
+        dismiss(animated: true, completion: nil)
+    }
+
+    func configureSlider() {
+        let tag = getUniqueIdentifier()
+        limitSlider.frame = calcFrame(by: tag)
+        limitSlider.minimumValue = 0
+        limitSlider.maximumValue = 200
+        limitSlider.value = 50
+        self.view.addSubview(limitSlider)
+
+        limitSlider.addTarget(self, action: #selector(changeSliderLable), for: .valueChanged)
+
+        sliderLable.frame = CGRect(x: Int(view.center.x) - 150,
+                                   y: 120 + 60 * tag,
+                                   width: 300, height: 40)
```

Comment:
It better to make struct or enum and store there constant numbers (where it's make sense)